src/js/models/panel_model.js

```diff
@@ -444,8 +455,20 @@
                     labs.push( lbl );
                 }
             }
+
+            // filter unique keys only to remove duplicates
+            var filtered_lbls_map = {}
+            for (var i=0; i<labs.length; i++) {
+                filtered_lbls_map[this.get_label_key(labs[i])] = i
+            }
+        
+            var filtered_lbls = [] 
+            _.each(Object.values(filtered_lbls_map), function(idx){
+                filtered_lbls.push(labs[idx])
+            })
```

Comment:
This is working fine for me now, thanks.
I don't want to sound too picky, but I had a slightly hard time understanding the code, and I'm also not sure that using `Object.values` will always preserve the order of the items (although it seems to be working).
I just wonder, since you're filtering, if it's clearer to use the `filter` function like this?

```
            // filter unique keys only to remove duplicates
            var keys = new Set(labs.map(lbl => this.get_label_key(lbl)));
            var filtered_lbls = labs.filter(lbl => keys.has(this.get_label_key(lbl)));
```

Apologies: I understand that this may not be a "better" way for everyone, and I don't want to make code review and PR submission any more time-consuming than it has to be. So please feel free to ignore this suggestion if it doesn't feel better for you!

Reply:
> I don't want to sound too picky,

Don't worry, you have much more experience than me on Javascript !

However, I tested your changes but duplicates are not always removed. If  `labs` contains duplicate labels, then duplicates are not removed.

Reply:
The snippet now works correctly. 
It is definitely more concise and I think it could be understandable quite easily (filtering based on the index of the first occurrence of the value)
So it should be ok 👍

Reply:
It also removes the dirty ``Object.values()``, which I don't really like as well